lib/responder.js

```diff
@@ -1156,6 +1157,17 @@ function operationResultStream() {
 function operationErrorListener(error) {
   /*jshint validthis:true */
   var operation = this;
+  if(operation.client.connectionParams.apiKey){
+    if(error.statusCode === 403 && operation.expiration <= (new Date())){
```

Comment:
Is the check on `expiration` to avoid an endless loop where a 403 is returned for some reason other than an expired token? Otherwise, I think it's better to just check for a 403 - that's what the Java Client and XCC are doing.

Reply:
Yes, it is to avoid loop.